### ISSUE TITLE
Request - Press button to resend

### ISSUE THREAD

**Issue by softy2k** [bartbutenaers/node-red-contrib-msg-resend#10]
Hello and thanks for this node.

I was wondering if you could make the msg resend if you press a button. In which case it does not resend automatically depending on the time, but only when you press the respective button.

As in the style of the button here: https://flows.nodered.org/node/node-red-contrib-reinject.

**Comment by bartbutenaers** [comment]
Hi @softy2k,
I had added a button - similar to that one - in a beta release of one of my other nodes in the past, but the reactions from the community were not really optimistic.  A button behind an input port, is indeed not really cosmetic attractive...

So I'm not going into that rabbit hole a second time.  Hope you understand ...
If I'm not mistaken, the feedback was to use an inject node that is wired to the input.  Something like this:

![image](https://user-images.githubusercontent.com/14224149/102538983-57b60a80-40ad-11eb-868a-feecf4cfbaeb.png)

Then you use the inject node as button, and you don't need to put two visual widgets on top of each other ...
Could this somehow - e.g. by adding some extra feature to the resend node - solve your feature request?
Bart

**Comment by softy2k** [comment]
Is OK.
As a functionality: the messages should go through the node, but their retransmission should be done only manually by pressing the reinject button.

Thanks in advance

**Comment by bartbutenaers** [comment]
Do I understand correctly that the resend-node should have different modes, and that you can switch the mode via input messages (e.g. triggered by a button):

![image](https://user-images.githubusercontent.com/14224149/102662134-805a0500-417e-11eb-9e56-0874f30486c5.png)

+ Messages 1 / 2 / 5 will simply be passed
+ Messages 3 / 4 will be resend N times

Is that correct?  Because that reinject-node would resend only message 2, but I cannot find any use cases for that in my mind ...

**Comment by softy2k** [comment]
I understand what you said, but that's not what I mean.

Basically, I would need something simpler.
All messages pass without any restriction. But, when you press the resend button, it only resends the last message. Without repeating it. If you press the button again, it sends the last message again.

In other words, pressing the resend button will relay the last incoming message. A kind of manual retry for the last incoming message. Just like https://flows.nodered.org/node/node-red-contrib-reinject., But unfortunately that node does not work well in certain situations, and I do not yet have the necessary knowledge to be able to improve it.

Maybe it would be simpler to create a new resend node, a light version.

I apologize for my English, maybe it's harder to understand what I'm trying to describe.

**Comment by bartbutenaers** [comment]
Hmm, I see that the author of the reinject node- doesn't has any activity on Github the last two years.  So most probably he doesn't maintain that node anymore.

Suppose we do it like this:
+ The resend node gets 2 states, as described in my image above.
+ When it is in "pass" state, you can send an input message with `msg.topic="resend_last"`, and then the last message will be resend N times (as specified in the config screen).

When you set N = 1 on the config screen, then the last message will be resend 1 times.

I think that would solve your specific use case, and I have a more ***general solution*** that can be used by other users.

Is that sufficient to solve your problem?

I would appreciate if you could test it thoroughly when it is ready, because I have VERY few spare time lately ...

**Comment by softy2k** [comment]
Is OK.

I will test, of course.

**Comment by bartbutenaers** [comment]
I have created a new branch "start-stop-resend" to this Github repository, which you can install like this (from within your .node-red folder):
```
npm install bartbutenaers/node-red-contrib-msg-resend#start-stop-resend
```
This branch allows you to:
+ `msg.resend_messages=false`:Tell this node to stop resending messages, which means the messages will just pass through this node.
+ `msg.resend_messages=true`:Tell this node to start resending messages.
+ `msg.resend_force=true`:Tell this node that this msg should always be resend, even if the resending mode has been stopped
+ `msg.resend_never=true`:Tell this node that this msg should never be resend, even if the resending mode has been started.  This means the msg will simply be passed through.
+ By default this node will start resending message already at startup/(re)deploy time, but you can optionally specify that it should start by passing messages (until a `msg.resend_messages=true` arrives):

   ![image](https://user-images.githubusercontent.com/14224149/102754233-eefcb580-436c-11eb-8f93-79a13cdcd208.png)

So don't confuse this with the existing `msg.resend_ignore=true`, which means *"never send this msg on the output"* (so no resend and no passthrough).  This option is mostly used for control messages (= messages purely used to send new config commands to this node), which should never be send to the output...

Here is an example flow that demonstrates these features:

![image](https://user-images.githubusercontent.com/14224149/102757192-3edd7b80-4371-11eb-922e-5adaad7ce4b8.png)
```
[{"id":"59598bca.9cca44","type":"msg-resend","z":"42b7b639.325dd8","interval":"1","intervalUnit":"secs","maximum":"4","bytopic":false,"clone":false,"firstDelayed":false,"addCounters":false,"waitForResend":true,"highRate":false,"outputCountField":"","outputMaxField":"","name":"","x":2150,"y":500,"wires":[["5b425903.0b78f8"]]},{"id":"9dd11c82.39201","type":"inject","z":"42b7b639.325dd8","name":"someValue","props":[{"p":"payload"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"someValue","payloadType":"str","x":1820,"y":500,"wires":[["59598bca.9cca44"]]},{"id":"47a07507.095fdc","type":"inject","z":"42b7b639.325dd8","name":"Stop resend","props":[{"p":"resend_ignore","v":"true","vt":"bool"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","x":1830,"y":800,"wires":[["59598bca.9cca44"]]},{"id":"5b425903.0b78f8","type":"debug","z":"42b7b639.325dd8","name":"Output","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"payload","targetType":"msg","statusVal":"","statusType":"auto","x":2310,"y":500,"wires":[]},{"id":"306ec40.c695c3c","type":"inject","z":"42b7b639.325dd8","name":"Stop resending","props":[{"p":"resend_messages","v":"false","vt":"bool"},{"p":"resend_ignore","v":"true","vt":"bool"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","x":1840,"y":880,"wires":[["59598bca.9cca44"]]},{"id":"83110a1.e2e25f8","type":"inject","z":"42b7b639.325dd8","name":"Start resending","props":[{"p":"resend_messages","v":"true","vt":"bool"},{"p":"resend_ignore","v":"true","vt":"bool"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","x":1840,"y":840,"wires":[["59598bca.9cca44"]]},{"id":"a9e21d46.01e66","type":"comment","z":"42b7b639.325dd8","name":"Control msg (never on output)","info":"","x":1860,"y":760,"wires":[]},{"id":"ca433ad7.568ca8","type":"inject","z":"42b7b639.325dd8","name":"alwaysResendValue","props":[{"p":"payload"},{"p":"resend_force","v":"true","vt":"bool"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"alwaysResendValue","payloadType":"str","x":1850,"y":600,"wires":[["59598bca.9cca44"]]},{"id":"3de2aebd.498ee2","type":"comment","z":"42b7b639.325dd8","name":"Always resend msg","info":"","x":1830,"y":560,"wires":[]},{"id":"1dd595bb.a40e7a","type":"comment","z":"42b7b639.325dd8","name":"Normal msg","info":"","x":1810,"y":460,"wires":[]},{"id":"c5167531.d47498","type":"inject","z":"42b7b639.325dd8","name":"neverResendValue","props":[{"p":"payload"},{"p":"resend_never","v":"true","vt":"bool"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"neverResendValue","payloadType":"str","x":1850,"y":700,"wires":[["59598bca.9cca44"]]},{"id":"7749ee82.98489","type":"comment","z":"42b7b639.325dd8","name":"Never resend msg","info":"","x":1830,"y":660,"wires":[]}]
```

The above functionality is general usable for all users.  However your request to resend the ***previous msg*** feeled a bit weird to add to the resend node: I need to start remembering messages, and clone them (because otherwise you get conflicts when resending the same message again).  Moreover this cloning of previous messages will have impact on performance for users with high data througput.   

Therefore I have decided to keep this functionality outside the node.  In the following example flow I use two function nodes:
+ One function node clones every message, and stores it on flow context memory.  Note that it does this topic-based, so you can (optionally) use different topics, and each topic will have its own previous messge.
+ One function node gets the previous message from flow context memory.

In the following example flow, the resend node is by default in pass-through node.  When you press the button, I reinject the last remembered message (with field `msg.resend_force=true` to make sure it is resend even in pass-through mode):
![image](https://user-images.githubusercontent.com/14224149/102758158-9defc000-4372-11eb-8fd5-ad640c261c17.png)

Hope to hear from you soon.
Bart

**Comment by softy2k** [comment]
I will test today.
Thanks for your work.

**Comment by bartbutenaers** [comment]
Before I forget: all these commands can (optionally) be ***topic dependent***.
In the following example flow, the 'topic dependent' checkbox is activated, which means that you can e.g. stop resending messages with `msg.topic="TOPIC_A"` (but resending messages with `msg.topic="TOPIC_B"` will keep going on):

![image](https://user-images.githubusercontent.com/14224149/102768486-8c61e480-4381-11eb-8edc-e411a9f4fcf2.png)
```
[{"id":"47a6e4d0.e9890c","type":"msg-resend","z":"42b7b639.325dd8","interval":"1","intervalUnit":"secs","maximum":"3","bytopic":true,"clone":false,"firstDelayed":false,"addCounters":false,"waitForResend":false,"highRate":false,"outputCountField":"","outputMaxField":"","name":"","x":2610,"y":1100,"wires":[["2c674f7c.9e4ce"]]},{"id":"24a67425.81772c","type":"inject","z":"42b7b639.325dd8","name":"someValue","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"TOPIC_A","payload":"someValue","payloadType":"str","x":2220,"y":1100,"wires":[["47a6e4d0.e9890c"]]},{"id":"b52429ac.93ed78","type":"inject","z":"42b7b639.325dd8","name":"Stop resend","props":[{"p":"resend_ignore","v":"true","vt":"bool"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"TOPIC_A","x":2230,"y":1220,"wires":[["47a6e4d0.e9890c"]]},{"id":"2c674f7c.9e4ce","type":"debug","z":"42b7b639.325dd8","name":"Output","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"payload","targetType":"msg","statusVal":"","statusType":"auto","x":2770,"y":1100,"wires":[]},{"id":"96f18e1f.ce573","type":"inject","z":"42b7b639.325dd8","name":"Stop resending","props":[{"p":"resend_messages","v":"false","vt":"bool"},{"p":"resend_ignore","v":"true","vt":"bool"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"TOPIC_A","x":2240,"y":1300,"wires":[["47a6e4d0.e9890c"]]},{"id":"2b7b5124.b01efe","type":"inject","z":"42b7b639.325dd8","name":"Start resending","props":[{"p":"resend_messages","v":"true","vt":"bool"},{"p":"resend_ignore","v":"true","vt":"bool"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"TOPIC_A","x":2240,"y":1260,"wires":[["47a6e4d0.e9890c"]]},{"id":"95cfd653.5efed8","type":"inject","z":"42b7b639.325dd8","name":"alwaysResendValue","props":[{"p":"payload"},{"p":"resend_force","v":"true","vt":"bool"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"TOPIC_A","payload":"alwaysResendValue","payloadType":"str","x":2250,"y":1140,"wires":[["47a6e4d0.e9890c"]]},{"id":"42c377f3.40f388","type":"inject","z":"42b7b639.325dd8","name":"neverResendValue","props":[{"p":"payload"},{"p":"resend_never","v":"true","vt":"bool"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"TOPIC_A","payload":"neverResendValue","payloadType":"str","x":2250,"y":1180,"wires":[["47a6e4d0.e9890c"]]},{"id":"66e43da2.e81494","type":"comment","z":"42b7b639.325dd8","name":"Topic A","info":"","x":2200,"y":1060,"wires":[]},{"id":"c8492c52.3081d","type":"inject","z":"42b7b639.325dd8","name":"someValue","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"TOPIC_B","payload":"someValue","payloadType":"str","x":2220,"y":1400,"wires":[["47a6e4d0.e9890c"]]},{"id":"b0393be5.bbcb98","type":"inject","z":"42b7b639.325dd8","name":"Stop resend","props":[{"p":"resend_ignore","v":"true","vt":"bool"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"TOPIC_B","x":2230,"y":1520,"wires":[["47a6e4d0.e9890c"]]},{"id":"f3318316.901fe","type":"inject","z":"42b7b639.325dd8","name":"Stop resending","props":[{"p":"resend_messages","v":"false","vt":"bool"},{"p":"resend_ignore","v":"true","vt":"bool"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"TOPIC_B","x":2240,"y":1600,"wires":[["47a6e4d0.e9890c"]]},{"id":"d469912e.eda73","type":"inject","z":"42b7b639.325dd8","name":"Start resending","props":[{"p":"resend_messages","v":"true","vt":"bool"},{"p":"resend_ignore","v":"true","vt":"bool"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"TOPIC_B","x":2240,"y":1560,"wires":[["47a6e4d0.e9890c"]]},{"id":"a7b7bb02.37bd88","type":"inject","z":"42b7b639.325dd8","name":"alwaysResendValue","props":[{"p":"payload"},{"p":"resend_force","v":"true","vt":"bool"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"TOPIC_B","payload":"alwaysResendValue","payloadType":"str","x":2250,"y":1440,"wires":[["47a6e4d0.e9890c"]]},{"id":"6a842c33.49fb84","type":"inject","z":"42b7b639.325dd8","name":"neverResendValue","props":[{"p":"payload"},{"p":"resend_never","v":"true","vt":"bool"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"TOPIC_B","payload":"neverResendValue","payloadType":"str","x":2250,"y":1480,"wires":[["47a6e4d0.e9890c"]]},{"id":"6dbe1d41.ae1bc4","type":"comment","z":"42b7b639.325dd8","name":"Topic B","info":"","x":2200,"y":1360,"wires":[]}]
```

**Comment by softy2k** [comment]
In the debug window we have the result of pressing buttons 1, 2 and 3.
I would say that it is not a good idea for the red border messages to go further, in flow. I think those messages should be filtered / stopped in the msg-resend node. Eventually, the texts "alwaysResendValue" and "neverResendValue" should appear as info (node.status)
![Repeat](https://user-images.githubusercontent.com/37790195/102910828-dcc16b00-4483-11eb-8234-8a3099b70637.png)

**Comment by bartbutenaers** [comment]
I don't think I have explained it well

   - I should have called it alwaysResendMsg: it is just an example of a
   message that should always be resend, even if you have told the node: don't
   resend messages
   - I should have called it neverResendMsg: it is just an example of a
   message that will never be resend (i.e. only passed through): even if you
   have told the node to resend messages.

This way you can resend specific messages (when other messages are being
passed through), and pass through specific messages (when other messages
are being resend).  So by adding a simple attribute to a message, you can
circumvent the mode (pass-throug or resending) of the node.

Hopefully it is a bit more clear now.

>

**Comment by softy2k** [comment]
I should have read more carefully: "Control msg (**never on output**). It's my fault, not yours.

**Comment by softy2k** [comment]
I tested the new functions, everything seems to be OK.

About this :


![Repeat2](https://user-images.githubusercontent.com/37790195/102918547-ca4d2e80-448f-11eb-8bfa-4fb75a733ea6.png)




I agree about : "Moreover this cloning of previous messages will have impact on performance for users with high data througput."
I understand why you decided to keep the clone and restore functions of the last message out of the node. My request is different from the essence of the node you created.

But keeping the clone and restore functions off the node is not a solution for me, because it already does this using **RED.util.cloneMessage (msg)**.
I understand your lack of time and I will probably try to modify / update the node-red-contrib-reinject node myself.

Maybe when you have time you will create something similar to node-red-contrib-reinject, whose only function is exactly what I need, but which unfortunately does not work properly in certain situations.

In other words, the new functions you added for the msg-resend node are welcome. I am convinced that they will be appreciated by others, not only by me. After all, adding new features is a welcome upgrade.

Thank you for your effort and time.

**Comment by bartbutenaers** [comment]
Thanks for testing!!!
Suppose I would remember the last msg, and resend that if msg you send a
`msg.resend_lastmsg=true`.  That is enough for you I assume?  I could skip
the cloning, because (like you say) it is already possible to specify that
it should be cloned during resending.
Will have a look, but we have movie evening with our boys.  So won't be
ready today...

**Comment by softy2k** [comment]
**`msg.resend_lastmsg=true`** is enough.

There is no rush for me, enjoy the movie.
Thanks again

**Comment by bartbutenaers** [comment]
Thanks.  It was a good movie.  I'm totally in the Christmas mood now :-)

I have a new version available on Github, so you should install it again from my branch using the above command.
This version allows you to inject a `msg.resend_last_msg=true`, to resend the last message again.

The following example flow allows you to inject two different messages, and then resend the last injected message:

![image](https://user-images.githubusercontent.com/14224149/102937174-d6de8080-44a9-11eb-8b2b-d24d5625c20b.png)
```
[{"id":"d0f5ca67.9191e8","type":"msg-resend","z":"42b7b639.325dd8","interval":"1","intervalUnit":"secs","maximum":"4","bytopic":false,"clone":false,"firstDelayed":false,"addCounters":false,"waitForResend":false,"highRate":false,"outputCountField":"","outputMaxField":"","name":"","x":2370,"y":220,"wires":[["b18a381c.2e5bf8"]]},{"id":"6c905f49.c7653","type":"inject","z":"42b7b639.325dd8","name":"someValue","props":[{"p":"payload"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"someValue","payloadType":"str","x":2100,"y":220,"wires":[["d0f5ca67.9191e8"]]},{"id":"b18a381c.2e5bf8","type":"debug","z":"42b7b639.325dd8","name":"Output","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"payload","targetType":"msg","statusVal":"","statusType":"auto","x":2530,"y":220,"wires":[]},{"id":"267d86d3.2e31ca","type":"inject","z":"42b7b639.325dd8","name":"Resend last message","props":[{"p":"resend_last_msg","v":"true","vt":"bool"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","x":2140,"y":320,"wires":[["d0f5ca67.9191e8"]]},{"id":"8d5a0592.2802e8","type":"inject","z":"42b7b639.325dd8","name":"anotherValue","props":[{"p":"payload"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"anotherValue","payloadType":"str","x":2110,"y":260,"wires":[["d0f5ca67.9191e8"]]}]
```

Note that this feature can also be ***topic dependent***, if you activate the checkbox in the config screen.  In the following example, you can inject both messages (one message per topic) and then resend the last message of each topic:

![image](https://user-images.githubusercontent.com/14224149/102937302-202ed000-44aa-11eb-8c09-d8acfe0463da.png)
```
[{"id":"2521d824.9cd678","type":"comment","z":"42b7b639.325dd8","name":"Topic A","info":"","x":2270,"y":660,"wires":[]},{"id":"8acdfc42.25be","type":"msg-resend","z":"42b7b639.325dd8","interval":"1","intervalUnit":"secs","maximum":"4","bytopic":true,"clone":false,"firstDelayed":false,"addCounters":false,"waitForResend":false,"highRate":false,"outputCountField":"","outputMaxField":"","name":"","x":2590,"y":700,"wires":[["b3a115ab.aec4f8"]]},{"id":"ecdde9d2.5b6218","type":"inject","z":"42b7b639.325dd8","name":"someValue","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"TOPIC_A","payload":"someValue","payloadType":"str","x":2280,"y":700,"wires":[["8acdfc42.25be"]]},{"id":"b3a115ab.aec4f8","type":"debug","z":"42b7b639.325dd8","name":"Output","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"payload","targetType":"msg","statusVal":"","statusType":"auto","x":2750,"y":700,"wires":[]},{"id":"3acd8ba.9c69c74","type":"inject","z":"42b7b639.325dd8","name":"Resend last topic A msg","props":[{"p":"resend_last_msg","v":"true","vt":"bool"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"TOPIC_A","x":2320,"y":740,"wires":[["8acdfc42.25be"]]},{"id":"ffa2c428.0fef28","type":"comment","z":"42b7b639.325dd8","name":"Topic B","info":"","x":2270,"y":800,"wires":[]},{"id":"b00d63c.ebcd6a","type":"inject","z":"42b7b639.325dd8","name":"anotherValue","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"TOPIC_B","payload":"anotherValue","payloadType":"str","x":2290,"y":840,"wires":[["8acdfc42.25be"]]},{"id":"58254d59.b624e4","type":"inject","z":"42b7b639.325dd8","name":"Resend last topic B msg","props":[{"p":"resend_last_msg","v":"true","vt":"bool"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"TOPIC_B","x":2330,"y":880,"wires":[["8acdfc42.25be"]]}]
```
Have fun with it!

**Comment by softy2k** [comment]
I did a few extra tests, everything seems to be OK.
In the coming days I will do other tests.

Finally, I got what I wanted :)

Once again, thank you for your work.

**Comment by bartbutenaers** [comment]
You are welcome!
Now you see that your persistence paid off ;-)
Would appreciate if you could do some extra tests in the coming days, because I have touched this node quite heavily...

**Comment by softy2k** [comment]
By pressing "**_Resend last message 2_**", **_Debug 2_** should have the text "**_Original input_**".

![Resend](https://user-images.githubusercontent.com/37790195/102941395-4741cd80-44bb-11eb-96d9-86871b4f5ba0.png)

`[{"id":"8514f10e.e030b8","type":"inject","z":"fc92f4ca.d11a28","name":"","props":[{"p":"payload"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"Original input","payloadType":"str","x":1690,"y":4020,"wires":[["d36ba8ff.e157"]]},{"id":"f6bcbdbe.d52ee","type":"debug","z":"fc92f4ca.d11a28","name":"Final","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"payload","targetType":"msg","statusVal":"","statusType":"auto","x":2770,"y":4020,"wires":[]},{"id":"eb4d3000.05767","type":"function","z":"fc92f4ca.d11a28","name":"New Payload","func":"msg.payload = 'Altered payload';\n\n\nreturn msg;","outputs":1,"noerr":0,"initialize":"","finalize":"","x":2610,"y":4020,"wires":[["f6bcbdbe.d52ee"]]},{"id":"d36ba8ff.e157","type":"msg-resend","z":"fc92f4ca.d11a28","interval":"1","intervalUnit":"msecs","maximum":"0","bytopic":false,"clone":true,"firstDelayed":false,"addCounters":false,"waitForResend":true,"highRate":false,"outputCountField":"","outputMaxField":"","name":"","x":1930,"y":4020,"wires":[["89629e3f.566008","243f9aac.a33fce"]]},{"id":"e89ea919.c14ad8","type":"inject","z":"fc92f4ca.d11a28","name":"Resend last message 1","props":[{"p":"resend_last_msg","v":"true","vt":"bool"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","x":1920,"y":3960,"wires":[["d36ba8ff.e157"]]},{"id":"89629e3f.566008","type":"debug","z":"fc92f4ca.d11a28","name":"Debug 1","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"payload","targetType":"msg","statusVal":"","statusType":"auto","x":2080,"y":4060,"wires":[]},{"id":"243f9aac.a33fce","type":"msg-resend","z":"fc92f4ca.d11a28","interval":"1","intervalUnit":"msecs","maximum":"0","bytopic":false,"clone":true,"firstDelayed":false,"addCounters":false,"waitForResend":true,"highRate":false,"outputCountField":"","outputMaxField":"","name":"","x":2330,"y":4020,"wires":[["eb4d3000.05767","fb98d1c5.428268"]]},{"id":"7b938293.9bee54","type":"inject","z":"fc92f4ca.d11a28","name":"Resend last message 2","props":[{"p":"resend_last_msg","v":"true","vt":"bool"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","x":2300,"y":3960,"wires":[["243f9aac.a33fce"]]},{"id":"fb98d1c5.428268","type":"debug","z":"fc92f4ca.d11a28","name":"Debug 2","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"payload","targetType":"msg","statusVal":"","statusType":"auto","x":2480,"y":4060,"wires":[]}]`

**It's late now, no hurry, there are days left.**

**Comment by bartbutenaers** [comment]
Morning @softy2k,
Should now be solved in the new version on github.
What has changed: when being in pass-through mode, the messages will now be cloned also (when "force clone" is activated).

Meanwhile I will update the readme file with a description of all new features.

**Comment by bartbutenaers** [comment]
The [readme ](https://github.com/bartbutenaers/node-red-contrib-msg-resend/tree/start-stop-resend) file is now hopefully up-to-date will all new features.  And if you reinstall this node, all example flows should be available via the Import-->Examples menu in the Node-RED flow editor...

**Comment by softy2k** [comment]
Everything seems to be OK now.

**Comment by bartbutenaers** [comment]
Hi @softy2k,
I have already published version ***1.0.0*** on NPM 12 days ago, but seems I have forgotten to let you know ...
Getting old ;-)
Bart
